merkle-tree/src/filter-controller.js

```diff
@@ -229,39 +228,51 @@ async function getFromBlock(db, contractName) {
     `Stats at restart, from the merkle-tree's mongodb: latestLeaf, ${latestLeaf}; blockNumber, ${blockNumber}`,
   );
 
-  if (blockNumber === undefined) {
-    let receipt;
-    let transactionHash = await utilsWeb3.getDeployedContractTransactionHash(contractName);
-    logger.info(` ${contractName} deployed transactionHash:  ${transactionHash}`);
+  if (!contractId) {
+    if (blockNumber === undefined) {
+      let receipt;
+      let transactionHash = await utilsWeb3.getDeployedContractTransactionHash(contractName);
+      logger.info(` ${contractName} deployed transactionHash:  ${transactionHash}`);
+
+      if (transactionHash) {
+        receipt = await utilsWeb3.getTransactionReceipt(transactionHash);
+        logger.info(`receipt: ${receipt}`);
+      }
+
+      blockNumber = receipt ? receipt.blockNumber : config.FILTER_GENESIS_BLOCK_NUMBER;
+      logger.warn(
+        `No filtering history found in mongodb, so starting filter from the contract's deployment block ${blockNumber}`,
+      );
+    }
 
-    if (transactionHash) {
-      receipt = await utilsWeb3.getTransactionReceipt(transactionHash);
-      logger.info(`receipt: ${receipt}`);
+  } else {
+    if (blockNumber === undefined) {
+      const contractInfo = await axios.get(process.env.CONTRACT_API_ENDPOINT + contractId);
```

Comment:
Does `CONTRACT_API_ENDPOINT` require an external endpoint to be always active? If so, how does it work?

Reply:
I believe so, I can't think of any other way to find the contract's deployment block only with its address. It is possible to fetch it from ETH Json-Rpc but only if we had the transaction hash of the deployment tx. And that's just adding extra steps.

Reply:
I see, I think calling the endpoint is fine for the starlight branch, but in main we want to keep Timber as a library which doesn't depend on anything external apart from web3. I agree that digging through the chain history to find the deployment block is just extra steps, but in future it might be good to edit `getDeployedContractTransactionHash` to take in the address and find the block some other way.

Reply:
We've decided to do all the fetching on the zapp's side and send the address to Timber (so Timber doesn't need to resolve it). (https://github.com/EYBlockchain/saas-transactional-ocm-zapps/tree/resolve-address)

The only issue to figure out is to how access the contract's deployment block from Timber without making any external calls.

Reply:
Update: We're working on fetching the block number on the zapp's side as well and passing that to Timber. That'll let us remove the last API call from the file mentioned above.

Reply:
Addressed in [0c41d29](https://github.com/EYBlockchain/timber/pull/133/commits/0c41d299b759f127662406caf1ff40c9d3c0089f)